include/eve/arch/x86/as_register.hpp

```diff
@@ -19,6 +19,9 @@ namespace eve
   struct logical;
   struct x86_128_;
   struct x86_256_;
+
+  template<typename Type, typename Cardinal, typename ABI>
+  struct as_register;
```

Comment:
Missing `as_register` primary definition. Should I have included smth instead?

Reply:
It's normally in the arch/cpu/as_register.hpp
Forward declaring is fine too.

Reply:
But it doesn't break any of our compilers even locally. The as_register should be used via the main arch/as_register.hpp as it includes everything it requires.